media/server/gstplayer/source/GstGenericPlayer.cpp

```diff
@@ -1457,16 +1457,42 @@ void GstGenericPlayer::setVolume(double targetVolume, uint32_t volumeDuration, f
     }
 }
 
-bool GstGenericPlayer::getVolume(double &volume)
+bool GstGenericPlayer::getVolume(double &currentVolume)
 {
     // We are on main thread here, but m_context.pipeline can be used, because it's modified only in GstGenericPlayer
     // constructor and destructor. GstGenericPlayer is created/destructed on main thread, so we won't have a crash here.
     if (!m_context.pipeline)
     {
         return false;
     }
-    volume =
-        m_gstWrapper->gstStreamVolumeGetVolume(GST_STREAM_VOLUME(m_context.pipeline), GST_STREAM_VOLUME_FORMAT_LINEAR);
+
+    GstElement *sink{getSink(MediaSourceType::AUDIO)};
+    if (!sink)
+    {
+        RIALTO_SERVER_LOG_ERROR("Failed to get fade volume property, sink is NULL");
+        return false;
+    }
+
+    if (m_glibWrapper->gObjectClassFindProperty(G_OBJECT_GET_CLASS(sink), "fade-volume"))
+    {
+        gdouble fadeVolume;
```

Comment:
Isn't the fadevolume property and int? also initialise  this variable to a invalid value